File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.py
#User function Template for python3

class Solution:
    ##Complete this function
    #Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self,arr):
        #Your code here
        ms=arr[0]
        maxend=arr[0]
        for i in range(1,len(arr)):
            maxend=max(maxend+arr[i],arr[i])
            ms=max(maxend,ms)
        return ms





#{ 
 # Driver Code Starts
#Initial Template for Python 3

import math


def main():
    T = int(input())
    while (T > 0):

        arr = [int(x) for x in input().strip().split()]

        ob = Solution()

        print(ob.maxSubArraySum(arr))
        print("~")
        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends

File: Difficulty: Medium/Kadane's Algorithm/kadanes-algorithm.java
class Solution {
    int maxSubarraySum(int[] arr) {
        // Your code here
        int ms=arr[0];
        int me=arr[0];
        for(int i=1;i<arr.length;i++){
            me=Math.max(me+arr[i],arr[i]);
            ms=Math.max(me,ms);
        }
        return ms;
    }
}
